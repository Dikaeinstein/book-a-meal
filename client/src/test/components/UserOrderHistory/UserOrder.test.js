import React from 'react';
import sweetalert from 'sweetalert';
import { shallow } from 'enzyme';
import { UserOrder }
  from '../../../components/UserOrderHistory/UserOrder';
import ConnectedUpdateOrder from '../../../components/UserOrderHistory/UpdateOrder';

jest.mock('sweetalert');

describe('<User />', () => {
  const order = {
    id: 11,
    quantity: 1,
    total: '4000',
    status: 'delivered',
    expired: false,
    mealId: 3,
    userId: 4,
    meal: {
      id: 3,
      name: 'Amala and Ewedu',
      price: '4000',
    },
    user: {
      id: 4,
      name: 'Chinonso Okorie',
    },
  };
  it('should call handleOpenModal', () => {
    const spy = jest.spyOn(UserOrder.prototype, 'handleOpenModal');
    const wrapper = shallow(<UserOrder order={order} modifyOrder={jest.fn()} />);
    const openModal = wrapper.find('.action-btn').first();
    openModal.simulate('click');
    expect(spy).toHaveBeenCalled();
    expect(wrapper.state().isModalOpen).toBeTruthy();
  });
  it('should call handleCloseModal', () => {
    const spy = jest.spyOn(UserOrder.prototype, 'handleCloseModal');
    const wrapper = shallow(<UserOrder
      order={order}
      modifyOrder={jest.fn()}
    />);
    wrapper.find(ConnectedUpdateOrder).props().closeModal();
    expect(spy).toHaveBeenCalled();
    expect(wrapper.state().isModalOpen).toBeFalsy();
  });
  it('should call handleCancel', () => {
    sweetalert.mockResolvedValue(Promise.resolve(true));
    const spy = jest.spyOn(UserOrder.prototype, 'handleCancel');
    const wrapper = shallow(<UserOrder
      order={{ ...order, status: 'cancelled' }}
      modifyOrder={jest.fn()}
    />);
    const cancelOrder = wrapper.find('.action-btn').last();
    cancelOrder.simulate('click');
    expect(spy).toHaveBeenCalled();
  });
  it('should call handleCancel', () => {
    sweetalert.mockResolvedValue(Promise.reject(new Error()));
    const spy = jest.spyOn(UserOrder.prototype, 'handleCancel');
    const wrapper = shallow(<UserOrder
      order={{ ...order, expired: false, status: 'pending' }}
      modifyOrder={jest.fn()}
    />);
    const cancelOrder = wrapper.find('.action-btn').last();
    cancelOrder.simulate('click');
    sweetalert.mockResolvedValue(Promise.reject());
    cancelOrder.simulate('click');
    expect(spy).toHaveBeenCalled();
  });
  it('should match snapshot', () => {
    const tree = shallow(<UserOrder order={order} modifyOrder={jest.fn()} />);
    expect(tree).toMatchSnapshot();
  });
});
