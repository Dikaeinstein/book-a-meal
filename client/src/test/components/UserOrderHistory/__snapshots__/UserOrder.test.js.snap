// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<User /> should match snapshot 1`] = `
<React.Fragment>
  <tr
    key="11"
  >
    <td>
      11
    </td>
    <td>
      Amala and Ewedu
    </td>
    <td>
      4000
    </td>
    <td>
      1
    </td>
    <td>
      4000
    </td>
    <td
      className="text-success"
    >
      delivered
    </td>
    <td>
      <div
        style={
          Object {
            "display": "flex",
          }
        }
      >
        <button
          className="action-btn"
          disabled={true}
          onClick={[Function]}
          style={
            Object {
              "backgroundColor": "#ccc",
              "cursor": "default",
            }
          }
          title="Update order"
          value="Edit"
        >
          Update
        </button>
        <button
          className="action-btn"
          disabled={true}
          onClick={[Function]}
          style={
            Object {
              "backgroundColor": "#ccc",
              "cursor": "default",
            }
          }
          title="Cancel order"
          value="Cancel"
        >
          Cancel
        </button>
      </div>
    </td>
  </tr>
  <Modal
    ariaHideApp={true}
    bodyOpenClassName="ReactModal__Body--open"
    closeTimeoutMS={150}
    contentLabel="Update order"
    isOpen={false}
    parentSelector={[Function]}
    portalClassName="ReactModalPortal"
    shouldCloseOnEsc={true}
    shouldCloseOnOverlayClick={true}
    shouldFocusAfterRender={true}
    shouldReturnFocusAfterClose={true}
    style={
      Object {
        "content": Object {
          "bottom": "auto",
          "left": "50%",
          "marginRight": "-50%",
          "minWidth": "20rem",
          "right": "auto",
          "top": "50%",
          "transform": "translate(-50%, -50%)",
          "width": "45%",
        },
        "overlay": Object {
          "backgroundColor": "rgba(0, 0, 0, 0.25)",
        },
      }
    }
  >
    <Button
      className="close"
      onClick={[Function]}
      value="Ã—"
    />
    <Connect(UpdateOrder)
      closeModal={[Function]}
      defaultOrder={
        Object {
          "expired": false,
          "id": 11,
          "meal": Object {
            "id": 3,
            "name": "Amala and Ewedu",
            "price": "4000",
          },
          "mealId": 3,
          "quantity": 1,
          "status": "delivered",
          "total": "4000",
          "user": Object {
            "id": 4,
            "name": "Chinonso Okorie",
          },
          "userId": 4,
        }
      }
    />
  </Modal>
</React.Fragment>
`;
